staticaddr/loopin/actions.go

```diff
@@ -0,0 +1,594 @@
+	// preventing us from continuing with a swap.
+	err = f.cfg.ValidateLoopInContract(
+		int32(f.loopIn.InitiationHeight), loopInResp.HtlcExpiry,
+	)
+	if err != nil {
+		return f.HandleError(err)
+	}
+
+	f.loopIn.HtlcCltvExpiry = loopInResp.HtlcExpiry
+	f.htlcServerNonces, err = toNonces(loopInResp.HtlcServerNonces)
+	if err != nil {
+		return f.HandleError(err)
+	}
+
+	// We need to defend against the server setting high fees for the htlc
+	// tx since we might have to sweep the timeout path. We maximally allow
+	// 20% of the swap value to be spent on fees.
+	f.loopIn.HtlcTxFeeRate = chainfee.SatPerKWeight(loopInResp.HtlcFeeRate)
+	fee := f.loopIn.HtlcTxFeeRate.FeeForWeight(f.loopIn.htlcWeight())
+
+	if fee > f.loopIn.TotalDepositAmount()/5 {
+		return f.HandleError(errors.New("fee is higher than 20% of " +
+			"sweep value"))
+	}
+
+	// Derive the sweep address for the htlc timeout sweep tx.
+	sweepAddress, err := f.cfg.WalletKit.NextAddr(
+		f.ctx, lnwallet.DefaultAccountName,
+		walletrpc.AddressType_TAPROOT_PUBKEY, false,
+	)
+	if err != nil {
+		return f.HandleError(err)
+	}
+	f.loopIn.HtlcTimeoutSweepAddress = sweepAddress
+
+	// Once the htlc tx is initiated, we store the loop-in in the database.
+	err = f.cfg.Store.CreateLoopIn(f.ctx, f.loopIn)
+	if err != nil {
+		return f.HandleError(err)
+	}
+
+	return OnHtlcInitiated
+}
+
+// SignHtlcTxAction is called if the htlc was initialized and the server
+// provided the necessary information to construct the htlc tx. We sign the htlc
+// tx and send the signatures to the server.
+func (f *FSM) SignHtlcTxAction(_ fsm.EventContext) fsm.EventType {
+	var err error
+
+	f.loopIn.AddressParams, err = f.cfg.AddressManager.GetStaticAddressParameters( //nolint:lll
+		f.ctx,
+	)
+	if err != nil {
+		return f.HandleError(err)
+	}
+
+	f.loopIn.Address, err = f.cfg.AddressManager.GetStaticAddress(f.ctx)
+	if err != nil {
+		return f.HandleError(err)
+	}
+
+	// Create a musig2 session for each deposit.
+	htlcSessions, clientHtlcNonces, err := f.loopIn.createMusig2Sessions(
+		f.ctx, f.cfg.Signer,
+	)
+	if err != nil {
+		return f.HandleError(err)
+	}
+	f.htlcMusig2Sessions = htlcSessions
+	f.htlcClientNonces, err = toNonces(clientHtlcNonces)
+	if err != nil {
+		return f.HandleError(err)
+	}
+
+	htlcTx, err := f.loopIn.createHtlcTx(f.cfg.ChainParams)
+	if err != nil {
+		return f.HandleError(err)
+	}
+
+	// Next we'll get our htlc tx signatures.
+	htlcSigs, err := f.loopIn.signMusig2Tx(
+		f.ctx, htlcTx, f.cfg.Signer, htlcSessions, f.htlcServerNonces,
+	)
+	if err != nil {
+		return f.HandleError(err)
+	}
+	f.htlcClientSigs = htlcSigs
+
+	// Push htlc tx sigs to server.
+	pushHtlcReq := &looprpc.PushStaticAddressHtlcSigsRequest{
+		SwapHash:         f.loopIn.SwapHash[:],
+		HtlcClientNonces: clientHtlcNonces,
+		HtlcClientSigs:   htlcSigs,
+	}
+	sigPushResp, err := f.cfg.StaticAddressServerClient.PushStaticAddressHtlcSigs( //nolint:lll
```

Comment:
nit: instead of adding `nolint` everywhere, we could just remove `StaticAddressServderClient` to `Server`